source/server/configuration_impl.cc

```diff
@@ -171,7 +172,30 @@ WatchdogImpl::WatchdogImpl(const envoy::config::bootstrap::v3::Watchdog& watchdo
   multikill_timeout_ =
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, multikill_timeout, 0));
   multikill_threshold_ = PROTOBUF_PERCENT_TO_DOUBLE_OR_DEFAULT(watchdog, multikill_threshold, 0.0);
-  actions_ = watchdog.actions();
+  auto actions = watchdog.actions();
+
+// Add abort_action if it's available on the given platform and killing is
+// enabled since it's more helpful than the default kill action.
+#ifndef WIN32
+  envoy::extensions::watchdog::abort_action::v3alpha::AbortActionConfig abort_config;
+  // Wait one second for the aborted thread to abort.
+  abort_config.mutable_wait_duration()->set_seconds(1);
+
+  if (kill_timeout > 0) {
+    auto abort_action_config = actions.Add();
```

Comment:
Perhaps an explicit type would be more readable here than `auto`, since the return type of `actions.Add()` is not obvious for first-time readers.

Reply:
Good idea. Done.